File: src/test/java/compiler/block/packages/PackageBlock.java
package compiler.block.packages;

public class PackageBlock extends Block {

    public String directory;

    public PackageBlock(String directory) {
        super(null, false, false);
        this.directory = directory;
    }

    @Override
    public void init() {

    }

    public String getName() {
        return null;
    }

    @Override
    public String getValue() {
        return null;
    }

    public String getType() {
        return null;
    }

    @Override
    public String getOpeningCode() {
        return null;
    }

    @Override
    public String getBodyCode() {
        return null;
    }

    @Override
    public String getClosingCode() {
        return null;
    }

}

File: src/main/scala/compiler/block/packages/PackageBlock.scala
package compiler.block.packages

import compiler.block.Block

class PackageBlock(var directory: String) extends Block(null, true, false) {

  def init() {}

  def getName: String = directory

  def getValue: String = null

  def getType: String = "package"

  def getOpeningCode: String = ""

  def getClosingCode: String = ""

  override def toString: String = "package: " + directory

}